plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -507,6 +507,24 @@ But that's okay, because these are all types we've been creating just now in Quo
 we should have globally unique names
 -}
 
+{- Note [Term/type argument mismatches]
+Given a term t and its type ty we can process them in parallel popping off arguments/function types.
+
+But we can end up with a mismatch:
+- We run out of arguments at the term level e.g. because we see something like `(\x -> \y -> y) 1`,
```

Comment:
These are both application. I wonder if it could be helpful to do something different for application?

Reply:
The question is: what? we'd have to actually reduce the application. Which means basically running the evaluator. Probably not worth it.